Source/Scene/TileBoundingS2Cell.js

```diff
@@ -0,0 +1,558 @@
+import BoundingSphere from "../Core/BoundingSphere.js";
+import Check from "../Core/Check.js";
+import ColorGeometryInstanceAttribute from "../Core/ColorGeometryInstanceAttribute.js";
+import defaultValue from "../Core/defaultValue.js";
+import GeometryInstance from "../Core/GeometryInstance.js";
+import Matrix4 from "../Core/Matrix4.js";
+import PerInstanceColorAppearance from "./PerInstanceColorAppearance.js";
+import Primitive from "./Primitive.js";
+import S2Cell from "../Core/S2Cell.js";
+
+var scratchCartographic = new Cartographic();
+
+/**
+ * A tile bounding volume specified as an S2 cell token with minimum and maximum heights.
+ * The bounding volume is a discrete oriented polytope.
+ * @alias TileBoundingS2Cell
+ * @constructor
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String} options.token The token of the S2 cell.
+ * @param {Number} [options.minimumHeight=0.0] The minimum height of the bounding volume.
+ * @param {Number} [options.maximumHeight=0.0] The maximum height of the bounding volume.
+ * @param {Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] The ellipsoid.
+ * @param {Boolean} [options.computeBoundingVolumes=true] True to compute the {@link TileBoundingS2Cell#boundingVolume} and
+ *                  {@link TileBoundingS2Cell#boundingSphere}. If false, these properties will be undefined.
+ *
+ * @private
+ */
+function TileBoundingS2Cell(options) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options", options);
+  Check.typeOf.string("options.token", options.token);
+  //>>includeEnd('debug');
+
+  this.s2Cell = S2Cell.fromToken(options.token);
+  this.minimumHeight = defaultValue(options.minimumHeight, 0.0);
+  this.maximumHeight = defaultValue(options.maximumHeight, 0.0);
+
+  var boundingPlanes = computeBoundingPlanes(
+    this.s2Cell,
+    this.minimumHeight,
+    this.maximumHeight
+  );
+
+  this._topPlane = boundingPlanes[0];
+  this._bottomPlane = boundingPlanes[1];
+  this._sidePlanes = boundingPlanes[2];
+
+  this._vertices = computeVertices(
+    boundingPlanes[0],
+    boundingPlanes[1],
+    boundingPlanes[2]
+  );
+
+  var points = new Array(8);
+
+  this.center = this.s2Cell.getCenter();
+
+  // Add center of cell.
+  points[0] = this.center;
+  scratchCartographic = Cartographic.fromCartesian(points[0]);
+  scratchCartographic.height = this.maximumHeight;
+  points[0] = Cartographic.toCartesian(scratchCartographic);
+  scratchCartographic.height = this.minimumHeight;
+  points[1] = Cartographic.toCartesian(scratchCartographic);
+  for (var i = 0; i <= 3; i++) {
+    scratchCartographic = Cartographic.fromCartesian(this.s2Cell.getVertex(i));
+    scratchCartographic.height = this.maximumHeight;
+    points[2 + i] = Cartographic.toCartesian(scratchCartographic);
+    scratchCartographic.height = this.minimumHeight;
+    points[2 + i + 1] = Cartographic.toCartesian(scratchCartographic);
```

Comment:
When I loaded the explicit tileset, I hit errors, I think the indexing should look like this:

```suggestion
    points[2 + 2 * i] = Cartographic.toCartesian(scratchCartographic);
    scratchCartographic.height = this.minimumHeight;
    points[2 + 2 * i + 1] = Cartographic.toCartesian(scratchCartographic);
```

Reply:
Instead of computing a `BoundingSphere` from an `OrientedBoundingBox`, I am now directly using the `BoundingSphere.fromPoints` function. This is how I was initially doing it, however, for some reason there were crashes in `BoundingSphere.js`. Either way, I can't replicate it now, and it seems to be working fine.